lib/utils/config.ts

```diff
@@ -68,6 +75,13 @@ export function generateBaseConfig(
 	return promise.tap(config => {
 		// os.getConfig always returns a config for an app
 		delete config.apiKey;
+
+		// merge sshKeys to config, when they have been specified
+		if(options.os && options.os.sshKeys) {
+			// Create config.os object if it does not exist
+			config.os = config.os ? config.os : {};
+			config.os.sshKeys = options.os.sshKeys;
```

Comment:
Just an observation: currently, the `config` obect returned by `getConfig` does not include an `os.sshKeys` field, but if did / when it  does, this code would override (replace) it with the value provided in the file specified in the command line. This is probably a reasonable behaviour. An alternative behaviour would be to add extra ssh keys, given that it is a list.

Reply:
You're right. When os.sshKeys are specified we should merge instead of override. Thanks for the catch, I'll change that.

On another note. What about the other params, that live inside the os object? 
Should I strive for a more general approach? Or do you think it is fine for now and extend this approach whenever it is needed?

Reply:
You mean, you could also merge or override other values like `os.network.*` or `os.udevRules.*`, right? I think it would be nice to have a generic merging approach where the `config.json` file given on the command line takes precedence (in case of value clashes), but not "too generic" because I think the CLI should only copy / merge "known" values, ignoring any "typos" or invalid fields. I suspect there is `config.json` format validation somewhere, if not on `balena-cli` then in other repos like maybe [balena-sdk](https://github.com/balena-io/balena-sdk) or [balena-supervisor](https://github.com/balena-io/balena-supervisor), which could be used in combination with a "generic merge" approach.

By the way, I've remembered something else, to keep in mind: except for the first time the device boots, the balena supervisor may overwrite `config.json` on the device to reflect the backend / API database. (I'm not sure about `os.sshKeys` specifically.) But I think your purpose with `balena os configure` is to configure the device before its first boot, so it should be OK.

Reply:
I took a look at the supervisor and the sdk, but I cannot seem to find a config.json format validation, so I think we would have to do it in the supervisor ourselves.

The sshKeys are copied from the config.json on first boot, see: https://github.com/balena-os/meta-balena/pull/1227/commits/6d2e43d4693208c8df9436ea1f96c844b4faeb75

Reply:
@pdcastro @mwolhert there is only validation of certain keys from config.json in the supervisor, we have no need to concern ourselves with the others. The supervisor also never touches `.os.sshKeys` (but as aside will touch `os.udevRules` very soon, just something to keep in mind), so it doesn't contain a schema for this.

My advice would be to not make this PR too complicated by trying to consider all of the future updates that may come, as these should be thoroughly tested by both the implementer and maintainer. I'm happy with this PR as it is, as it solves the problem nicely, and shouldn't have any side effects as far as I can see. This could definitely be a rabbit hole to fall down, so let's avoid it entirely. 

Thanks again for the submission Michel!

Reply:
@CameronDiver Thanks for sharing your view on this. I'll keep this pr as it is then and do not develop some kind of generic merging and validation approach.